jbmc/src/java_bytecode/README.md

```diff
@@ -6,6 +6,21 @@ This module provides a front end for Java.
 \section java-bytecode-conversion-section Overview of conversion from bytecode to codet
 
 To be documented.
+\subsection java-bytecode-lowering-to-goto Lowering to GOTO
+
+The Java language contains high level programming concepts like virtual functions and throw/catch semantics. We need to rewrite these in terms of other, more fundamental operations in order to analyse it. This operation is referred to as "lowering".
+
+The following lowering operations are done on java bytecode after converting into a basic codet representation.
+
+- \ref java-bytecode-runtime-exceptions "Add runtime exceptions"
```

Comment:
Hmm - I consider it in the same bucket since the JVM detects null pointer access and throws an exception - we lower that "high level feature" of the JVM into a fundamental operation of `if(a == null)`, what would you call it? 

Remove_nondet is kind of a weird one since it doesn't exist in Java bytecode or the JVM - it is a CPROVER construct right?

Reply:
I think we should carefully distinguish between lowering and instrumentation. The former is a rewrite of the code (supposedly semantics-preserving), while the latter will add or modify behaviour.